microsoft/AzureTRE
# Airlock processor handles request Submission

# Fixes #1836 

## What is being addressed

Adding support for the StatusChangedEvent in the Airlock processor (for submission - import and export)
Consolidated the blob created queues into a single SB topic
Wired all TF scripts
Added support for the blobCreatedEvent in the AL processor

## How is this addressed

- Added code to extract the msg from the status changed queue
- If data copy is required, extract the source and dest params, and issue a copy command
- added unit tests

How to test:
1. Create container in the import external storage. Give it a name: e.g. 123
2. Put a single file
3. Place this message in the status_changed queue

{ "data": {
    "request_id" : "123",
    "type":"import", 
    "status":"submitted" , 
    "workspace_id":"124" 
    }
}

Check the in progress storage account , it should contain a new container with the same file.
Check the step result queue, you should find a step result event which states the request is now in-progress (transitioned from submitted)
